relayer/src/foreign_client.rs

```diff
@@ -1226,10 +1226,6 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
             // Check for misbehaviour according to the specific source chain type.
             // In case of Tendermint client, this will also check the BFT time violation if
             // a header for the event height cannot be retrieved from the witness.
-            //
-            // FIXME: This returns error if the update event contains expired client state.
-            // This can happen even if the latest client state is unexpired, but the
-            // update event is from earlier.
             let misbehavior = self
```

Comment:
Note to reviewers: this is where the `Err` leaked out of `check_misbehaviour` when the trusted state was expired;  we were interpreting that as an abnormal failure in misbehavior check, but it was not a failure. I changed `check_misbehaviour` to return `Ok(None)` now, so I also removed the FIXME.

Reply:
I wouldn't say it was an "abnormal failure", it was a warning. imo we should still return warn and not Ok. Maybe make the message more clear to indicate that we are stopping misbehaviour due to expired states. Or return Ok but bail out if misbehavior runs for all states is in progress. The problem in this branch is that at start we now continue to go back through all expired consensus states. Recent gaia versions use SDK versions that prune expired consensus states so it is not an issue with these. For gaia v5.0.x this problem shows. 

Also the current debug is a bit confusing, why would misbehavior be Ok with expired states?

Reply:
Also not sure why this was marked as `FIXME`?

Reply:
Should be fixed now, thanks for catching that Anca! https://github.com/informalsystems/ibc-rs/pull/2156/commits/ba9831f834403071152ca1c9273a6e31aeba74e2 Please weigh in.